model/project_vars.go

```diff
@@ -632,59 +647,108 @@ func fullSyncToParameterStore(ctx context.Context, vars *ProjectVars, pRef *Proj
 		}
 	}
 
-	after := vars
-
-	// Since this is fully syncing all the project vars to Parameter Store,
-	// always perform comparisons against a clean state where there are no
-	// parameters.
-	before := &ProjectVars{}
-
-	// There's no variables to delete because any pre-existing ones were already
-	// deleted above.
-	varsToUpdate, _ := getProjectVarsDiff(before, after)
-
-	paramMappingsToUpdate, err := after.upsertParameters(ctx, before.Parameters, varsToUpdate)
-	if err != nil {
-		return errors.Wrapf(err, "upserting %d parameters for project vars", len(varsToUpdate))
-	}
-
-	updatedParamMappings := getUpdatedParamMappings(before.Parameters, paramMappingsToUpdate, nil)
+	return insertParameterStore(ctx, vars, pRef, isRepoRef)
+}
 
-	if _, err := db.Upsert(ProjectVarsCollection,
-		bson.M{projectVarIdKey: after.Id},
-		bson.M{"$set": bson.M{projectVarsParametersKey: updatedParamMappings}},
+func (projectVars *ProjectVars) Insert() error {
+	if err := db.Insert(
+		ProjectVarsCollection,
+		projectVars,
 	); err != nil {
-		return errors.Wrap(err, "updating parameter mappings for project vars after full sync")
+		return err
 	}
 
-	vars.Parameters = updatedParamMappings
+	// This has to be done after inserting the initial document because it
+	// upserts the project vars doc. If this ran first, it would cause the DB
+	// insert to fail due to the ID already existing.
+	ctx, cancel := context.WithTimeout(context.Background(), defaultParameterStoreAccessTimeout)
+	defer cancel()
 
-	coll := ProjectRefCollection
-	if isRepoRef {
-		coll = RepoRefCollection
+	ref, isRepoRef, err := projectVars.findProjectRef()
+	grip.Error(message.WrapError(err, message.Fields{
+		"message":    "could not get project ref to check if Parameter Store is enabled for project; assuming it's disabled and falling back to using the DB",
+		"op":         "Insert",
+		"project_id": projectVars.Id,
+		"epic":       "DEVPROD-5552",
+	}))
+	var isPSEnabled bool
+	if ref != nil {
+		isPSEnabled, err = isParameterStoreEnabledForProject(ctx, ref)
```

Comment:
I have a preference to add `if ref == nil { return false }` to the top of `isParameterStoreEnabledForProject` to avoid the need for nil checks, but it's not really important and I'll leave it up to you